pkg/scheduler/core/generic_scheduler.go

```diff
@@ -648,6 +648,12 @@ func PrioritizeNodes(
 			}(i, priorityConfig)
 		} else {
 			results[i] = make(schedulerapi.HostPriorityList, len(nodes))
+			for index, node := range nodes {
```

Comment:
Instead of this hot loop that goes over all nodes for each priority function and assigns `results[i][index]`, add
`results[i][index] = schedulerapi.HostPriority{ Host:  node.Name, Score: 0, }` to the body of `if err != nil` at line 663.

Reply:
Good point.

And could be more performant: `results[i][index].Host = node.Name` b/c previously the HostPriority slice was already initialized with a length.

Reply:
To move the initialize function, I delete the `return` and init the rest `results` when error happens.

Reply:
Yup, when I first looked at it my understanding was, either we need to lazily update the slice when we catch errors or initialize the slice at the starting before returning. I think the advantage of initializing the slice before catching error is the chance of us getting into the error block is almost 0 but yes this comes at the expense of performance(which we could do by lazily updating the slice). I am fine with this approach too. Just wanted to make sure, all of us are on the same page.